### DISCUSSION TITLE
Class from @layer breaks compilation when used with @apply

### DISCUSSION THREAD

**Discussion by nemanjam** [tailwindlabs/tailwindcss#7343] [Help] [answered]
I made test case on Stackblitz:

https://stackblitz.com/edit/github-tanduc-rd6ymd?file=styles%2Ftest.css

In `styles/globals.css` I defined a class within `base` layer:

```css
@layer base {
  .test-class-base {
    @apply bg-blue-400;
  }
}
```
I used that class in `styles/test.css` with `@apply`

```css
.test-css {
  @apply test-class-base;
}
```

I used `test-css` class in `pages/index.tsx

```html
<h1 className="text-6xl font-bold test-css">my title</h1>
```

```
Syntax error: /home/projects/github-tanduc-rd6ymd/styles/test.css The `test-class-base` class does not exist. If `test-class-base` is a custom class, make sure it is defined within a `@layer` directive.
```

If `test-class-base` is used directly in jsx it compiles, if it is used in css with `@apply` it breaks compilation. How to fix this?

**Comment by gynekolog** [comment]
There exist a few open issues but without a solution for now:
https://github.com/tailwindlabs/tailwindcss/issues/7205, https://github.com/tailwindlabs/tailwindcss/issues/5126

**Comment by adamwathan** [comment, marked as answer]
This is essentially what's described in the docs here:

https://tailwindcss.com/docs/functions-and-directives#using-apply-with-per-component-css

You are running Tailwind twice — once on `globals.css` and once on `test.css` and those two runs are isolated with no knowledge of each other.

The best fix in your case is to use `postcss-import` and import `test.css` inside of `globals.css` and just process the one file. Working with Tailwind is generally a lot smoother when you just have a single global CSS file instead of lots of individual files that you process independently.

Here are some docs on setting up build-time imports: https://tailwindcss.com/docs/using-with-preprocessors#build-time-imports

**Reply by nemanjam** [reply]
Yes, that's it, all .scss files must be imported in a single index.scss and not multiple scss files into .tsx.
